src/main/java/org/springframework/social/bitbucket/connect/BitBucketAdapter.java

```diff
@@ -46,10 +51,23 @@ public final void setConnectionValues(BitBucket api, ConnectionValues values) {
 
     @Override
     public final UserProfile fetchUserProfile(BitBucket api) {
-        BitBucketUser user = api.userOperations().getUserWithRepositories()
-                .getUser();
+        BitBucketUser user = api.userOperations().getUserWithRepositories().getUser();
+        String primaryEmail = null;
+        try {
+            List<BitBucketEmailAddress> emailAddresses = api.usersOperations()
+                    .emailsOperations().getListOfUserEmailAddresses(user.getUsername());
+            for (BitBucketEmailAddress emailAddress : emailAddresses) {
+                if (emailAddress.getPrimary()) {
+                    primaryEmail = emailAddress.getEmail();
+                }
+            }
+        } catch (HttpClientErrorException clientError) {
+            if (!HttpStatus.FORBIDDEN.equals(clientError.getStatusCode())) {
```

Comment:
Is there a reason to ignore other HTTP errors? I would prefer to let the exceptions bubble up. The client of that API probably expect the full profile to be loaded by this method. At least we should log a warning for exceptions that are hidden.